Add Binary Search in JavaScript

izexi opened the pull request:
## I Am Adding a New Code Snippet in an Existing Language

- [x] I fixed #2287 
- [x] I named the pull request using `Add {PROJECT} in {LANGUAGE}` format
- [x] I created/updated the language README (see [contributing documentation][contributing-readme])
  - [x] I added the sample program name to the README
  - [ ] I added fun facts (i.e. debut developer, typing, etc.)
  - [ ] I added reference link(s) to the README
  - [ ] I added solution citations when necessary (see [plagiarism][contributing-plagiarism])


[renegade-coder]: https://therenegadecoder.com/
[contributing-plagiarism]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#plagiarism
[contributing-new-project]: https://github.com/TheRenegadeCoder/sample-programs/blob/contributing/.github/CONTRIBUTING.md#requirements-for-a-new-project
[contributing-readme]: https://github.com/TheRenegadeCoder/sample-programs/blob/contributing/.github/CONTRIBUTING.md#create-readmes
[contributing-tests-in-detail]: https://github.com/TheRenegadeCoder/sample-programs/blob/contributing/.github/CONTRIBUTING.md#tests-in-detail
[contributing]: ../CONTRIBUTING.md
[sample-programs-project-list]: https://sample-programs.therenegadecoder.com/projects/
[contributing-modifications]: https://github.com/TheRenegadeCoder/sample-programs/blob/contributing/.github/CONTRIBUTING.md#modifying-existing-code-snippets

jrg94 reviewed (changes requested):
Looks great, but I don't think this ended up getting tested. It looks like our JavaScript naming convention is hyphen.

izexi commented:
> Looks great, but I don't think this ended up getting tested. It looks like our JavaScript naming convention is hyphen.

@jrg94 I have renamed the files to use hyphens in https://github.com/TheRenegadeCoder/sample-programs/pull/2288/commits/b2265492b9c9e12a98a8637d388af2f3c3d9c3f3 and also renamed the Longest Palindromic Substring's file name to match the abbreviation on the test file.

jrg94 reviewed (approved):
Nice! Thanks for the help.
